hcloud/instances_util.go

```diff
@@ -92,6 +92,14 @@ func getRobotServerByID(c robot.Client, id int, node *corev1.Node) (*hrobotmodel
 
 	// check whether name matches - otherwise this server does not belong to the respective node anymore
 	if server.Name != node.Name {
+		i.recorder.Eventf(
+			node,
+			corev1.EventTypeWarning,
+			"PossibleNodeDeletion",
+			"Might be deleted by node-lifecycle-manager due to name mismatch; Node name \"%s\" differs from Robot name \"%s\"",
```

Comment:
Tip: You can use `%q` to get a quoted string:

```suggestion
			"Might be deleted by node-lifecycle-manager due to name mismatch; Node name %q differs from Robot name %q",
```